test/jdk/jdk/crac/jdwp/JdwpTransportTest.java

```diff
@@ -140,21 +144,31 @@ public void test() throws Exception {
             process.input().flush();
 
             process.waitForSuccess();
+        } catch (TimeoutException e) {
+            System.err.println("reader.isRunning()=" + reader.isRunning());
+            CracProcess.printThreadDump(process.pid());
+            CracProcess.dumpProcess(process.pid());
+            throw e;
         } finally {
-                for (String line = errReader.readLine(); null != line; line = errReader.readLine()) {
-                    System.err.println(line);
+            try {
+                final long timeoutMillis = 1000;
+                while (true) {
+                    System.err.println("APP STDERR: " + errReader.readLine(timeoutMillis));
                 }
-                process.destroyForcibly();
+            } catch (TimeoutException e) {
+                // do nothing
+            }
+            process.destroyForcibly();
         }
     }
 
     @Override
     public void exec() throws Exception {
-        System.out.println(STARTED);
+        System.out.println(STARTED + ", pid=" + ProcessHandle.current().pid());
+        System.out.flush();
```

Comment:
AFAIK `System.out` is auto-flushed when `\n` is written, so `println` should be sufficient